src/jit/compiler.h

```diff
@@ -297,6 +297,208 @@ enum RefCountState
+//  a BasicBlock is sent to "setDumperStartAt" so we can dump all
+//  the "VariableLiveRange"s from that one.
+//  After we dump all the "VariableLiveRange"s we call "reset" with
+//  the "liveRangeList" to set the barrier to nullptr or the last
+//  "VariableLiveRange" if it is opened.
+//  If no "VariableLiveRange" was edited/added during block,
+//  the iterator points to the end of variable's LiveRangeList.
+//
+class LiveRangeDumper
+{
+private:
+    LiveRangeListIterator m_StartingLiveRange; // Iterator to the first edited/added position
+                                               // during actual block code generation. If last
+                                               // block had a closed "VariableLiveRange" (with
+                                               // a valid "m_EndEmitLocation") and not changes
+                                               // were applied to variable liveness, it points
+                                               // to the end of variable's LiveRangeList.
+    bool m_hasLiveRangestoDump;                // True if a live range for this variable has been
+                                               // reported from last call to EndBlock
+
+public:
+    LiveRangeDumper(const LiveRangeList* liveRanges);
+
+    ~LiveRangeDumper(){};
+
+    // Make the dumper point to the last "VariableLiveRange" opened or nullptr if all are closed
+    void resetDumper(const LiveRangeList* list);
+
+    //  Make "LiveRangeDumper" instance points the last "VariableLifeRange" added so we can
+    // starts dumping from there after the actual "BasicBlock"s code is generated.
+    void setDumperStartAt(const LiveRangeListIterator liveRangeIt);
+
+    // Return an iterator to the first "VariableLiveRange" edited/added during the current
+    // "BasicBlock"
+    LiveRangeListIterator getStartForDump() const;
+
+    // Return whether at least a "VariableLiveRange" was alive during the current "BasicBlock"'s
+    // code generation
+    bool hasLiveRangesToDump() const;
+};
+#endif // DEBUG
+
+//--------------------------------------------
+//
+// VariableLiveDescriptor: This class persist and update all the changes
+//  on the variable home of a variable. It has an instance of "LiveRangeList"
+//  and methods to report the start/end of a VariableLiveRange.
+//
+// Notes:
+//  It has an instance of "LiveRangeList"
+//  and methods to report the start/end of a VariableLiveRange.
+//
+class VariableLiveDescriptor
+{
+private:
+    LiveRangeList* m_VariableLiveRanges; // the variable homes of this variable
+
+#ifdef DEBUG
+    LiveRangeDumper* m_VariableLifeBarrier;
+#endif // DEBUG
+
+public:
+    VariableLiveDescriptor(CompAllocator allocator);
+
+    ~VariableLiveDescriptor();
+
+    bool hasVariableLiveRangeOpen() const;
+
+    size_t getAmountLiveRanges() const;
+
+    const LiveRangeList* getLiveRanges() const;
+
+    void startLiveRangeFromEmitter(CodeGenInterface::siVarLoc varHome, emitter* _emitter) const;
```

Comment:
What's up `_emitter`? It's not common to use variables prefixed with `_` in the JIT.

Reply:
You are right, I am changing it. Is it common to have a class name starting with a lowercase letter? In this case the class name is "emitter".

Reply:
I have found places where it is used the name "emit", that one seems to be the best for now.

Reply:
> Is it common to have a class name starting with a lowercase letter?

Fortunately no, it's not common. Let's just say naming a class `emitter` rather than `Emitter` was a bad idea but we have to live with it because nobody's probably accept a PR changing the name, too much churn just for cosmetic reasons :)